### PR TITLE
Allow custom ignored namespaces in backtrace processor

### PR DESCRIPTION
**Pull request by adam-ja**
The backtrace processor adds the file, line, class, and function
responsible for triggering a log to the event extras.

By default, classes in the backtrace under the `Zend\Log` namespace are
ignored so that we can see the actual application code that called the
`log()`.

However, there might also be cases when we want to ignore other
namespaces so that we get the true origin. The example I came across
that prompted this patch was `Psr\Log`, which appears when the
`PsrLoggerAdapter` wraps the underlying Zend Logger.

This patch allows us to add our own custom ignored namespaces in
addition to the default `Zend\Log`.

### CODE REVIEW

**Comment by weierophinney** [comment]
This looks like a great feature!

Can you please also add documentation (under the `doc/book/` tree) detailing the new methods and options?

Thanks!

**Comment by adam-ja** [comment]
Docs added - sorry for the delay.

There's really nothing to say about `getIgnoredNamespaces()` as there's probably not a great deal of use for it beyond testing/debugging.

File: src/Processor/Backtrace.php [hunk 1]
```diff
@@ -59,6 +59,31 @@ public function process(array $event)
         return $event;
     }
 
+    /**
+     * Add a custom namespace to be ignored from the backtrace
+     *
+     * @param string $namespace
+     * @return self
+     */
+    public function addIgnoredNamespace($namespace)
```
Comment (Xerkus) [inline src/Processor/Backtrace.php]: Can it be changed to constructor injection?
I see no reason to change value after processor was created, as new namespaces to ignore should really be added by adding more configured processors

**Comment by michalbundyra** [comment]
Thanks, @adam-ja!

**Pull request merged**